plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -62,6 +62,52 @@ gs_flatpak_transaction_set_no_deploy (FlatpakTransaction *transaction, gboolean
 	g_object_notify (G_OBJECT (self), "no-deploy");
 }
 
+/* Sets installed app(s) back to installed state. Since libflatpak can return apps as updatable
+ * for re-installing a runtime if its is detected that the runtime was missing at the first place.
+ *
+ * We can determine whether is GsApps is only being updated due to a missing runtime, by checking
+ * if the current operation's ref is the GsApp's runtime and also the GsApp is already deployed.
+ */
+static void
+set_installed_app_state_if_missing_runtime_is_installed (FlatpakTransaction          *transaction,
+							  FlatpakTransactionOperation *operation)
+{
+	GsFlatpakTransaction *self = GS_FLATPAK_TRANSACTION (transaction);
+	FlatpakInstallation *installation = flatpak_transaction_get_installation (transaction);
+	const gchar *op_ref = flatpak_transaction_operation_get_ref (operation);
+	g_autoptr(GList) apps = NULL;
+
+	if (g_str_has_prefix (op_ref, "app/"))
+		return;
+
+	apps = g_hash_table_get_values (self->refhash);
+	for (GList *l = apps; l != NULL; l = l->next) {
+		GsApp *app = l->data;
+		GsApp *app_runtime;
+		g_autofree gchar *app_runtime_ref = NULL;
+
+		app_runtime = gs_app_get_runtime (app);
+		app_runtime_ref = gs_flatpak_app_get_ref_display (app_runtime);
```

Comment:
Don't we need `if (app_runtime == NULL) continue;` here?

Reply:
There is a `app_runtime !=NULL ` just below this

Reply:
But is it safe to call `gs_flatpak_app_get_ref_display(NULL)`?